internal/reorder/plan.go

```diff
@@ -0,0 +1,71 @@
+package reorder
+
+import (
+	"emperror.dev/errors"
+	"github.com/aviator-co/av/internal/git"
+	"github.com/aviator-co/av/internal/meta"
+)
+
+// CreatePlan creates a reorder plan for the stack rooted at rootBranch.
+func CreatePlan(repo *git.Repo, tx meta.ReadTx, rootBranch string) ([]Cmd, error) {
+	var cmds []Cmd
+
+	branchNames := []string{rootBranch}
+	branchNames = append(branchNames, meta.SubsequentBranches(tx, rootBranch)...)
+
+	for _, branchName := range branchNames {
+		branch, _ := tx.Branch(branchName)
+
+		branchCmd := StackBranchCmd{
+			Name: branchName,
+		}
+		// Need to figure out the upstream commit to figure out the list of
+		// commits associated with this branch.
+		var upstreamCommit string
+		// TODO: would be nice to show the user whether or not the branch is
+		// 		already up-to-date with the parent.
+		if branch.Parent.Head != "" {
+			branchCmd.Parent = branch.Parent.Name
+			upstreamCommit = branch.Parent.Head
+		} else {
+			trunkCommit, err := repo.MergeBase(&git.MergeBase{
```

Comment:
Hm.

Wonder why this is not using `branch.Parent.Head`. Can you add a comment on this?

Reply:
It's documented on `BranchState.Head`:
```
	// The commit SHA of the parent's latest commit. This is used when syncing
	// the branch with the parent to identify the commits that belong to the
	// child branch (since the HEAD of the parent branch may change).
	// This will be unset if Trunk is true.
```

A lot of that stuff shows its age from the very beginning of `av` though. `BranchState` should probably be like `ParentState` for clarity and `Trunk`/`Head` should be combined (since `Trunk` is `true` iff `Head` is empty -- which is annoying since the default is that `Trunk` is `false` and `Head` is empty which is an illegal state).

Generally it's just confusing wherever its used and seems to require keeping a lot of that info just in your head. I don't really want to add a comment here specifically because then we just end up copying the field documentation everywhere, but at some point I would like to change it to make it a bit more intuitive.

Reply:
Also, to clarify, it's unset when `Trunk` is true because we can always derive it as `git merge-base <trunk> <branch>` -- we only need to store the parent head for non-root branches because the parent can change/be amended, at which point we can no longer tell what commits belong to the old version of the parent and the branch itself unless we do this.

Reply:
Sorry. Talked offline. I somehow read the condition the opposite way.